### PR TITLE
Add documentation entry for start build inputs

### PR DESCRIPTION
**Pull request by priitlatt**
Add new documentation entry to YAML -> Advanced configuration -> Build inputs.

Added article contains the following information:
- describes what "start build inputs" are,
- how to set up build inputs in YAML workflows,
- how to use build inputs when starting builds manually and via API.

**TODOs**
- [ ] Release the feature in Codemagic

### CODE REVIEW

File: content/knowledge-codemagic/build-inputs.md [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+    inputs:
+      name:
+        description: Who is greeted?
+        required: false
+        default: Codemagic
+    scripts:
+      - echo "Hello, ${{ inputs.name }}"
+{{< /highlight >}}
+
+## Starting Builds with Inputs
```
Comment (helinanever) [inline content/knowledge-codemagic/build-inputs.md:27]:
```suggestion
## Starting builds with Inputs
```

File: content/knowledge-codemagic/build-inputs.md [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+
+To successfully start a build, all required inputs must be specified. For optional inputs (those with required: false), the provided default value will be used if no value is specified when the build is started.
+
+{{<notebox>}}
+**Note**: Builds will fail if invalid values are provided for inputs (strings for numbers inputs, undefined choice options, etc.) or values are missing for required inputs.
+{{</notebox>}}
+
+### Starting Builds Manually via Codemagic UI
```
Comment (helinanever) [inline content/knowledge-codemagic/build-inputs.md:35]:
```suggestion
### Starting builds manually via Codemagic UI
```

File: content/knowledge-codemagic/build-inputs.md [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+
+When starting a build via the Codemagic UI, you will automatically be prompted to enter the inputs. Optional inputs are pre-filled with default values, but all required inputs must be manually entered before the build can be started.
+
+### Starting Builds Using REST API
```
Comment (helinanever) [inline content/knowledge-codemagic/build-inputs.md:39]:
```suggestion
### Starting builds using REST API
```

File: content/knowledge-codemagic/build-inputs.md [hunk 4]
```diff
@@ -0,0 +1,242 @@
...
+
+To start a build using the REST API, values for all required inputs must be included in the `POST` request payload. For more detailed information on starting builds with inputs using the REST API, refer to the section [below](#specify-inputs-when-starting-builds-with-api). Optional inputs can be omitted from the request payload; their default values will be used instead.
+
+### Starting Builds Using Webhook Events
```
Comment (helinanever) [inline content/knowledge-codemagic/build-inputs.md:43]:
```suggestion
### Starting builds using webhook events
```

**Review by helinanever — COMMENTED** [review]
Very nice! Some nits related to capitalization for consistency. Let's wait until the UI is finalized to understand if we need to change anything about starting builds.

File: content/knowledge-codemagic/build-inputs.md (hunk 4 above)
Comment (priitlatt) [reply content/knowledge-codemagic/build-inputs.md]: Resolved in [ef40ff7](https://github.com/codemagic-ci-cd/codemagic-docs/pull/2537/commits/ef40ff791410c359c31f08359bb843fd96004fa8).

File: content/knowledge-codemagic/build-inputs.md (hunk 3 above)
Comment (priitlatt) [reply content/knowledge-codemagic/build-inputs.md]: Resolved in [ef40ff7](https://github.com/codemagic-ci-cd/codemagic-docs/pull/2537/commits/ef40ff791410c359c31f08359bb843fd96004fa8).

File: content/knowledge-codemagic/build-inputs.md (hunk 2 above)
Comment (priitlatt) [reply content/knowledge-codemagic/build-inputs.md]: Resolved in [ef40ff7](https://github.com/codemagic-ci-cd/codemagic-docs/pull/2537/commits/ef40ff791410c359c31f08359bb843fd96004fa8).

File: content/knowledge-codemagic/build-inputs.md (hunk 1 above)
Comment (priitlatt) [reply content/knowledge-codemagic/build-inputs.md]: Resolved in [ef40ff7](https://github.com/codemagic-ci-cd/codemagic-docs/pull/2537/commits/ef40ff791410c359c31f08359bb843fd96004fa8).

File: content/knowledge-codemagic/build-inputs.md [hunk 5]
```diff
@@ -0,0 +1,249 @@
...
+
+## Overview
+
+Build inputs are customizable parameters you can define within your workflow to make it more adaptable to different scenarios. With build inputs, you can create a single workflow and run it with different configurations by providing the values for inputs when starting a build for the workflow. For example, you can use build inputs to determine whether to build the workflow for test or release purposes or which app flavor to build. This eliminates the need to create multiple similar workflows with specific hardcoded values, making the workflow more reusable and dynamic.
+
+Inputs are workflow-specific and are defined in `codemagic.yaml` under the `inputs` mapping (see the [example](#minimal-example) below). The started workflow receives specified input values in the `inputs` context, i.e. `${{ inputs.inputId }}` is replaced with the value passed to input with identifier `inputId`.
+
+### Minimal example
+
+{{< highlight yaml "style=paraiso-dark">}}
```
Comment (mikhail-tokarev) [inline content/knowledge-codemagic/build-inputs.md:15]: will be nice to add short intro: "This example configures one input with `ID=name`. The name input is not required and will default to `Codemagic` value. 

- because `name` looks like reserved word, but in reality this is user-defined input ID.
- is it correct example? is it possible to use `inputs` in scripts? I was thinking we need to create environment variable first 🤔

File: content/knowledge-codemagic/build-inputs.md [hunk 6]
```diff
@@ -0,0 +1,249 @@
...
+        default: Codemagic
+    scripts:
+      - echo "Hello, ${{ inputs.name }}"
+{{< /highlight >}}
+
+## Starting builds with Inputs
+
+All inputs must be specified to successfully start a build, either by providing a `default` value in the YAML configuration, or giving a one-off value when starting the build. 
+
+{{<notebox>}}
```
Comment (mikhail-tokarev) [inline content/knowledge-codemagic/build-inputs.md:30]: let's delete everything about using Rest API with Inputs.
Better to replace with information about webhooks and scheduled builds.

File: content/knowledge-codemagic/build-inputs.md [hunk 7]
```diff
@@ -0,0 +1,249 @@
...
+**Note**: Builds will fail if invalid values are provided (strings for numbers inputs, undefined choice options, etc.) or values are missing.
+{{</notebox>}}
+
+### Starting builds manually via Codemagic UI
+
+When starting a build via the Codemagic UI, you will automatically be prompted to enter the inputs. Inputs that have predefined default values will be pre-filled with those values from configuration file. All other inputs must be manually entered before the build can be started.
+
+Not entering anything for a string input will result in an empty string, i.e. `""`. For other input types an actual value which matches the requested type must be entered. 
+
+### Starting builds using REST API
```
Comment (mikhail-tokarev) [inline content/knowledge-codemagic/build-inputs.md:40]: let's remove this section

File: content/knowledge-codemagic/build-inputs.md [hunk 8]
```diff
@@ -0,0 +1,249 @@
...
+
+To start a build using the REST API, values for inputs that do not define `default` must be included in the `POST` request payload. The inputs which have default value declared in YAML configuration can be omitted from request payload, in that case the specified default will be used.  
+
+Given values must be in accordance with the input definitions, that is:
+- type of the value must match with the input type (numeric values for number inputs, truth values for boolean inputs and textual values for string inputs),
+- value for choice input must be included in the options list. 
+
+For more detailed information on starting builds with inputs using the REST API, refer to the section [below](#specify-inputs-when-starting-builds-with-api). Optional inputs can be omitted from the request payload; their default values will be used instead.
+
+### Starting builds using webhook events
```
Comment (mikhail-tokarev) [inline content/knowledge-codemagic/build-inputs.md:50]: what happens if no default value provided?

**Review by mikhail-tokarev — CHANGES_REQUESTED** [review]
- my main question about examples and is it really possible to reference inputs from scripts? just want to confirm
- let's exclude rest api part to keep focus on UI

**Comment by helinanever** [comment]
Removed the API section, and updated the minimal example and the section about using webhook events. 

Referencing inputs from scripts works fine.
>my main question about examples and is it really possible to reference inputs from scripts? just want to confirm

File: content/knowledge-codemagic/build-inputs.md [hunk 9]
```diff
@@ -0,0 +1,200 @@
...
+      name:
+        description: Who is greeted?
+        default: Codemagic
+    scripts:
+      - echo "Hello, ${{ inputs.name }}"
+{{< /highlight >}}
+
+## Starting builds with Inputs
+
+All inputs must be specified to successfully start a build, either by providing a `default` value in the YAML configuration, or giving a one-off value when starting the build.
```
Comment (mikhail-tokarev) [inline content/knowledge-codemagic/build-inputs.md:30]: this text duplicated and sounds better in the next section.

I suggest to remove this header and move 2 next sections to 1 level up:
1. Start builds with Inputs manually
2. Start builds with Inputs automatically

File: content/knowledge-codemagic/build-inputs.md [hunk 10]
```diff
@@ -0,0 +1,200 @@
...
+
+When starting a build via the Codemagic UI, you will automatically be prompted to enter the inputs. Inputs that have predefined default values will be prefilled with those values from configuration file. All other inputs must be manually entered before the build can be started.
+
+{{<notebox>}}
+**Note on inputs without default values:**
+
+Not entering anything for a string input will result in an empty string, i.e. `""`. Boolean inputs default to `false` in the UI. For other input types, an actual value that matches the requested type must be entered.
+{{</notebox>}}
+
+### Starting builds using webhook events
```
Comment (mikhail-tokarev) [inline content/knowledge-codemagic/build-inputs.md:46]: the header doesn't reflect scheduled or other ways without UI interaction - see my suggest above

File: content/knowledge-codemagic/build-inputs.md [hunk 11]
```diff
@@ -0,0 +1,200 @@
...
+
+Only workflows that do not require user input for values can be started with webhook events. If you want to use Git events or scheduled builds to automatically trigger builds for workflows with inputs, ensure that all inputs in those workflows have default values. Otherwise, the build will fail due to undefined inputs.
+
+## YAML schema for inputs
+
+Build inputs are defined in `codemagic.yaml` as a mapping `workflow.<workflow_id>.inputs` where keys are input IDs and values are inputs that have the following fields.
```
Comment (mikhail-tokarev) [inline content/knowledge-codemagic/build-inputs.md:52]:
```suggestion
Build inputs are defined in `codemagic.yaml` as a mapping `workflows.<workflow_id>.inputs` where keys are input IDs and values are inputs that have the following fields.
```

File: content/knowledge-codemagic/build-inputs.md [hunk 12]
```diff
@@ -0,0 +1,200 @@
...
+
+Defines the data type of the input parameter. Input values for the `choice` type are resolved to strings and must be defined in the `options` field. Values for inputs with types `boolean` and `number` are persisted as booleans and integers or floating point numbers respectively, instead of converting them to strings as long as they are not directly used in string interpolations.
+
+### `options`
+
+Provide a list of values as options for the `choice` input. Values are all implicitly cast to strings. **Required if type is `choice` and prohibited otherwise**.
+
+### `default`
+
+Provide a default value for the input parameter. If `type` is choice, then it must be one of the defined `options`, otherwise value type must match with the `type` definition.
```
Comment (mikhail-tokarev) [inline content/knowledge-codemagic/build-inputs.md:70]:
```suggestion
Provide a default value for the input parameter. If type is `choice`, then it must be one of the defined `options`, otherwise value type must match with the `type` definition.
```

File: content/knowledge-codemagic/build-inputs.md [hunk 13]
```diff
@@ -0,0 +1,200 @@
...
+
+## Examples
+
+### String inputs
```
Comment (mikhail-tokarev) [inline content/knowledge-codemagic/build-inputs.md:74]: Pass input values to script commands

File: content/knowledge-codemagic/build-inputs.md [hunk 14]
```diff
@@ -0,0 +1,200 @@
...
+    inputs:
+      name:
+        description: Who is greeted?
+    scripts:
+      - echo "Hello, ${{ inputs.name }}"
+{{< /highlight >}}
+
+As in the above example no `default` value is provided, then name must be specified when starting builds for this workflow, or otherwise it will be left blank.
+
+### Boolean inputs
```
Comment (mikhail-tokarev) [inline content/knowledge-codemagic/build-inputs.md:90]: Using inputs to conditionally run scripts

File: content/knowledge-codemagic/build-inputs.md [hunk 15]
```diff
@@ -0,0 +1,200 @@
...
+      app_store_connect: MY_ASC_KEY
+    environment:
+      ios_signing:
+        distribution_type: ${{ inputs.distributionType }}
+        bundle_identifier: com.example.app
+    scripts:
+      - xcode-project use-profiles
+      - flutter build ipa --debug --export-options-plist "${HOME:?}/export_options.plist"
+{{< /highlight >}}
+
```
Comment (mikhail-tokarev) [inline content/knowledge-codemagic/build-inputs.md:168]:
```suggestion
```

File: content/knowledge-codemagic/build-inputs.md [hunk 16]
```diff
@@ -0,0 +1,200 @@
...
+### Number inputs
```
Comment (mikhail-tokarev) [inline content/knowledge-codemagic/build-inputs.md:169]: Using inputs for publishing

File: content/rest-api/builds.md [hunk 17]
```diff
@@ -19,14 +19,15 @@ APIs for managing builds are currently available for developers to preview. Duri
...
+| `appId`       | `string` | **Required.** The application identifier.                                                                                                               |
+| `workflowId`  | `string` | **Required.** The workflow identifier as specified in YAML file.                                                                                        |
+| `branch`      | `string` | Optional. The branch name. Either `branch` or `tag` is **required**.                                                                                    |
+| `tag`         | `string` | Optional. The tag name. Either `branch` or `tag` is **required**.                                                                                       |
+| `environment` | `object` | Optional. Specify environment variables, variable groups, and software versions to override or define in workflow settings.                             |
+| `inputs`      | `object` | Optional. Specify values for build inputs. Read more about from [here](/knowledge-codemagic/build-inputs/#specify-inputs-when-starting-builds-with-api) |
```
Comment (mikhail-tokarev) [inline content/rest-api/builds.md:29]: imho better to revert this changes. There is no schema information or example.

**Review by mikhail-tokarev — CHANGES_REQUESTED** [review]

File: content/knowledge-codemagic/build-inputs.md (hunk 15 above)
Comment (helinanever) [reply content/knowledge-codemagic/build-inputs.md:164]: Why would we not want an example of choice input?

File: content/knowledge-codemagic/build-inputs.md [hunk 18]
```diff
@@ -0,0 +1,194 @@
...
+
+Inputs are workflow-specific and are defined in `codemagic.yaml` under the `inputs` mapping (see the [example](#minimal-example) below). The started workflow receives specified input values in the `inputs` context, i.e. `${{ inputs.inputId }}` is replaced with the value passed to input with identifier `inputId`.
+
+### Minimal example
+
+This example configures one input with the ID `name`. Unless given another value when starting a build, `name` defaults to `Codemagic`.
+
+{{< highlight yaml "style=paraiso-dark">}}
+workflows:
+  greetings:
```
Comment (kevin-suhajda) [inline content/knowledge-codemagic/build-inputs.md:19]: I would change the workflow name to `ios` or `ios-workflow`.

File: content/rest-api/builds.md (hunk 17 above)
Comment (helinanever) [reply content/rest-api/builds.md:29]: I've added yaml schema. Should we still remove this?

File: content/knowledge-codemagic/build-inputs.md (hunk 18 above)
Comment (helinanever) [reply content/knowledge-codemagic/build-inputs.md]: I've changed it to `my-workflow` so it's clear it's the workflow name.

File: content/rest-api/builds.md (hunk 17 above)
Comment (mikhail-tokarev) [reply content/rest-api/builds.md:29]: yes, better to remove. it's not about yaml but request schema

File: content/knowledge-codemagic/build-inputs.md [hunk 19]
```diff
@@ -0,0 +1,220 @@
...
+
+### `default`
+
+Provide a default value for the input parameter. If type is `choice`, then it must be one of the defined `options`, otherwise value type must match with the `type` definition.
+
+## Examples
+
+Below are some example use cases for leveraging different types of build inputs in your workflows.
+
+### Pass input values to script commands (strings)
```
Comment (mikhail-tokarev) [inline content/knowledge-codemagic/build-inputs.md:94]:
```suggestion
### Using input values in scripts
```

File: content/knowledge-codemagic/build-inputs.md [hunk 20]
```diff
@@ -0,0 +1,220 @@
...
+      app_store_connect: MY_ASC_KEY
+    scripts:
+      - name: Run tests
+        script: xcode-project run-tests --project "project.xcodeproj" --scheme "App"
+        test_report: build/ios/test/*.xml
+        when:
+          condition: ${{ inputs.runTests }}
+      - ./setup_code_signing.sh
+      - xcode-project build-ipa --project "project.xcodeproj" --scheme "app"
+      - echo "Hello, ${{ inputs.name }}"
```
Comment (mikhail-tokarev) [inline content/knowledge-codemagic/build-inputs.md:136]:
```suggestion
```

**Review by mikhail-tokarev — APPROVED** [review]
LGTM
